modules/ximgproc/src/ED.h

```diff
@@ -0,0 +1,129 @@
+/**************************************************************************************************************
+* Edge Drawing (ED) and Edge Drawing Parameter Free (EDPF) source codes.
+* Copyright (C) 2016, Cuneyt Akinlar & Cihan Topal
+* E-mails of the authors: cuneytakinlar@gmail.com, cihant@anadolu.edu.tr
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
```

Comment:
Unfortunately, we can't accept code under GPL license.

BTW, License file from upstream repository contains MIT license: https://github.com/CihanTopal/ED_Lib/blob/7f171db0321de67b12ee3d0558b0b1d60122000d/LICENSE